src/components/item/sharing/shareButton/CreateItemMembershipForm.tsx

```diff
@@ -91,41 +90,24 @@ const CreateItemMembershipForm = ({
         BUILDER.SHARE_ITEM_FORM_INVITATION_EMAIL_EXISTS_MESSAGE,
       );
     }
-    return null;
+    return true;
   };
 
-  const onChangePermission: ItemMembershipSelectProps['onChange'] = (e) => {
-    setInvitation({
-      ...invitation,
-      permission: e.target.value as PermissionLevel,
-    });
-  };
-
-  const handleShare = async () => {
-    // not good to check email for multiple invitations at once
-    const invitationError = checkForInvitationError(invitation);
-
-    if (invitationError) {
-      return setError(invitationError);
-    }
-
+  const handleShare = async (data: Inputs) => {
     let returnedValue;
     try {
       await shareItem({
         itemId,
         invitations: [
           {
-            email: invitation.email,
-            permission: invitation.permission,
+            email: data.email,
+            permission: data.permission,
           },
         ],
       });
 
       // reset email input
-      setInvitation({
-        ...invitation,
-        email: '',
-      });
+      reset({ email: '', permission: PermissionLevel.Read });
```

Comment:
I think you would not need to reset the form if the dialog is not persisted. You can try to instantiate the Dialog Content in a different component than the `Dialog` wrapper, so the state that controls the dialog is not inside the dialog content and the dialog can be destroyed and your form should be reset correctly.